app/models/github/issue_label.rb

```diff
@@ -10,20 +10,48 @@ def self.for_type(type, val)
     return unless TYPES.include?(type)
     return unless TYPES[type].include?(val)
 
-    "#{type_prefix(type)}#{val}"
+    "x:#{type}/#{val}"
   end
 
-  def of_type?(type)
-    name.start_with?(Github::IssueLabel.type_prefix(type))
+  def of_type?(type_to_check)
+    return unless value && type && namespace
+
+    namespace == :exercism &&
+      type == type_to_check &&
+      TYPES.include?(type_to_check) &&
+      TYPES[type_to_check].include?(value)
   end
 
   memoize
   def value
-    name.split('/').second.to_sym
+    return unless components
+
+    components[:value]&.to_sym
+  end
+
+  memoize
+  def type
+    return unless components
+
+    components[:type]&.to_sym
   end
 
-  def self.type_prefix(type)
-    "x:#{type}/"
+  memoize
+  def namespace
+    return unless components
+
+    case components[:namespace]
+    when 'x'
+      :exercism
+    else
+      :unknown
+    end
+  end
```

Comment:
And this could probably be `exercism_namespace?`which is just `components&.fetch(:namespace) == :exercism`. As we don't use `:unknown`, we probably don't need to have it here.